addons/pkg-npm/jaxrs/src/main/java/org/commonjava/indy/pkg/npm/jaxrs/NPMContentAccessResource.java

```diff
@@ -38,27 +41,35 @@
 import javax.ws.rs.core.Response;
 import javax.ws.rs.core.StreamingOutput;
 import javax.ws.rs.core.UriInfo;
+import java.nio.file.Paths;
 
 import static org.commonjava.indy.IndyContentConstants.CHECK_CACHE_ONLY;
 import static org.commonjava.indy.pkg.npm.model.NPMPackageTypeDescriptor.NPM_CONTENT_REST_BASE_PATH;
 import static org.commonjava.indy.pkg.npm.model.NPMPackageTypeDescriptor.NPM_PKG_KEY;
 
 @Api( value = "NPM Content Access and Storage", description = "Handles retrieval and management of NPM artifact content. This is the main point of access for NPM users." )
 @Path( "/api/content/npm/{type: (hosted|group|remote)}/{name}" )
+@ApplicationScoped
 public class NPMContentAccessResource
                 implements PackageContentAccessResource
 {
 
     private final Logger logger = LoggerFactory.getLogger( getClass() );
 
+    private static final String CROSS = "-";
+
+    private static final String PATH_SPLITER = "/";
+
+    private static final String PACAGE_TGZ = "package.tgz";
```

Comment:
@ligangty 
Real storage path for one **.tgz package is /{package}/{version}/package.tgz, and for any retrieve path of this package, it will be mapping to /{package}/-/{package}-{version}.tgz.

Why here to do the path mapping?
The metadata/package GET path s following what npm does, and for the PUT path, I make that sort of regular format like:
/package/package.json
/package/version/package.json
/package/version/package.tgz
which will be pure api to follow, and it's also what npm does after refering to npm storage in your local.